content/doc/developer/testing/index.adoc

```diff
@@ -597,8 +597,50 @@ To use them in your plugin, please find documentation here:
 * link:https://github.com/jenkins-infra/pipeline-library#runbenchmarks[Running benchmark through Jenkinsfile]
 
 == Further Pipeline Testing
+
+Sometimes you might need to test more sophisticated things, as restarting Jenkins and want to verify that the changes are still present.
+
 === Testing Durable Pipeline Steps
-TODO: RestartableJenkinsRule.
+
+If you want to ensure that your plugin behaves correctly after a restart, there are methods to help you.
+`RestartableJenkinsRule` is part of the Jenkins testing framework and provides an easy way to simulate restarting Jenkins in your tests without actually having to restart the Jenkins instance.
+You can use the `then()` method to run one Jenkins session and then shut down.
+The following example shows how it can be used with a `FreeStyleJob` run which is verified after a Jenkins restart:
```

Comment:
I lack fundamental skills in the use of `RestartableJenkinsRule`, so I'm not a great reviewer of this pull request.  My apologies in advance for mistakes that I may make in the review process.

Since the example uses a `FreeStyleProject`, I think that the section heading should be changed from "Testing Durable Pipeline Steps" to "Testing Jenkins Restart".  The example looks reasonable to me as a way to show that a `FreeStyleProject` build status persisted across a restart.

I think the job type used in the test should be referenced using the phrase "Freestyle job", the phrase "Freestyle project", or the Java class name "`FreeStyleProject`".  The first two forms are used elsewhere in the www.jenkins.io documentation and the third is used in the example source code.
```suggestion
The following example shows how it can be used with a `FreeStyleProject` build which is verified after a Jenkins restart:
```